src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -110,6 +110,13 @@ ucs_status_t uct_rdmacm_cm_ep_conn_notify(uct_ep_h ep)
               uct_rdmacm_cm_ep_str(cep, ep_str, UCT_RDMACM_EP_STRING_LEN),
               cep->id, rdmacm_cm, rdmacm_cm->ev_ch);
 
+    if (cep->flags & (UCT_RDMACM_CM_EP_FAILED |
+                      UCT_RDMACM_CM_EP_GOT_DISCONNECT)) {
+        remote_data.field_mask = 0;
+        uct_rdmacm_cm_ep_set_failed(cep, &remote_data, cep->status);
```

Comment:
why is this needed? we should have already called disconnect callback when set the FAILED flag?
should just return error in this case

Reply:
connect notify and error callback can be invoked from different threads, so this is needed to report error by err callback immediately

Reply:
what is the purpose of calling uct_rdmacm_cm_ep_set_failed() when FAILED flag is set, if uct_rdmacm_cm_ep_set_failed() not doing anything if FAILED flag is set (line 78)?

Reply:
to avoid code duplication since we have protection for failed state and move the EP to failed state on disconnect event during wireup